File: PART2/ch08/service/api/src/main/java/org/delivery/api/config/health/HealthOpenApiController.java
package org.delivery.api.config.health;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.delivery.api.common.rabbitmq.Producer;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RequiredArgsConstructor
@Slf4j
@RestController
@RequestMapping("/open-api")
public class HealthOpenApiController {

    @GetMapping("/health")
    public void health(){
        log.info("health call");
    }
}

File: PART3/ch03/ch03_springcloudapigateway/ch03_03/service/api/src/main/kotlin/org/delivery/api/config/health/HealthOpenApiController.kt
package org.delivery.api.config.health

import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/open-api")
class HealthOpenApiController {

    private val logger : Logger = LoggerFactory.getLogger(this.javaClass)

    @GetMapping("/health")
    fun health(): String{
        logger.info("health call")
        return "pong"
    }

}